include/bm/bm_sim/data.h

```diff
@@ -38,6 +38,17 @@ namespace bm {
 
 using bignum::Bignum;
 
+// DataRepresentation is used to represent data for log_msg.
+// if log_msg is to support a data type, the class for that type
+// should inherit this class and override show method.
+// For now the subclasses are Data, Header and List.
+class DataRepresentation {
```

Comment:
I think the convention in the rest of the code is to use the `Iface` suffix for pure abstract classes. So I would suggest `StringRepresentationIface` and maybe the `show` method should be renamed to `get_string_repr`. There is already a `get_string_repr` method in `Data` and your `show` implementation is calling it so why not use it directly?

Reply:
I've implemented the change you suggested and I've fixed format issues. I will upload the code once I do further analysis based on the comments that follow and when I add unit test.